test/test-applications/integrations/Samples.MongoDB/Program.cs

```diff
@@ -18,7 +18,9 @@ private static string Host()
             return Environment.GetEnvironmentVariable("MONGO_HOST") ?? "localhost";
         }
 
-        public static void Main(string[] args)
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        public static async Task Main(string[] args)
```

Comment:
What is this change for?

Reply:
We could probably change `RunAsync(collection, newDocument).Wait();` to `await RunAsync()`. I'm not sure why we were waiting synchronously on that task before

Reply:
The change isn't anything to do with the async/sync nature - it's because Datadog.Trace.dll is no longer in the bin folder. With the `void` method, the app tries to load the dll _before_ the profiler is loaded. With the async, the loader has time to run and load Datadog.Trace first. It's hacky, but it works 😉

Reply:
I didn't see there was an actual async method in there, with that I should be able to remove the warning 😄 
Note that the workaround shouldn't be needed when we switch to the new loader: since it uses module initializers, it should be able to run before the Main function.

Reply:
Can we add a comment explaining that making Main async is a workaround?

Reply:
@lucaspimentel I don't mind but the async Main becomes "legitimate" with https://github.com/DataDog/dd-trace-dotnet/pull/1726/commits/c2ddd2e6b969ca2a7d4e8506f2cbf947460a9c3e so I don't know if the comment is still appropriate (= the other samples work because they already have async Main, should I add a comment to every sample project?)

Reply:
No need for the comment anymore, thanks!